src/services/outliningElementsCollector.ts

```diff
@@ -81,21 +81,22 @@ namespace ts.OutliningElementsCollector {
             }
 
             if (!result[1]) {
-                const span = createTextSpanFromBounds(sourceFile.text.indexOf("//", currentLineStart), lineEnd);
+                const span = createTextSpanFromBounds(sourceFile.text.indexOf("//", currentLineStart), result[3] ? lineEnd - 1 : lineEnd);

```

Comment:
As we discussed offline, I think it would be preferable to keep the handling of `\r` and `\n` consistent.

Reply:
No longer needed with current changes.